src/operator/nn/fully_connected-inl.h

```diff
@@ -176,124 +162,80 @@ class FullyConnectedOp : public Operator {
     linalg_gemm(grad, wmat, gdata, false, false, s, req[fullc::kData]);
   }
 
+  static FullyConnectedOp &get_op(const FullyConnectedParam& param) {
+    static thread_local FullyConnectedOp op;
+    op.Init(param);
+    return op;
+  }
+
```

Comment:
What's the reasoning behind the static/thread_local object? What calls this? What's the guarantee that this won;t ever be called more than once on the same thread, or is there a reason it doesn't matter?  Seems like it would call Init() more than once in this case?

Reply:
static thread_local is to create an object once for each thread. Keeping this statement in a function is to help reference the object from different functions (i.e., forward and backward functions).

Init() has to be called every time the forward/backward function is invoked. Because the current operator interface is stateless. This is the best way to initialize the state for the followup operations in the forward/backward.

Reply:
The state is just the params, right?  It's almost always better to make an object/call stateless if possible, so since you have the params in-hand when thi function is calld, just pass it to the function as a parameter and work directly off of it without doing a copy or all of the fancy thread_local stuff.

Reply:
I agree that we can make activation completely stateless.